openpype/hosts/blender/api/pipeline.py

```diff
@@ -122,12 +124,23 @@ def set_start_end_frames():
     scene.render.resolution_y = resolution_y
 
 
+def set_base_file_unit_scale():
+    project = os.environ.get("AVALON_PROJECT")
+    settings = get_project_settings(project)
+
+    unit_scale = settings.get("blender").get("base_file_unit_scale")
+
+    bpy.context.scene.unit_settings.scale_length = unit_scale
+
+
 def on_new():
     set_start_end_frames()
+    set_base_file_unit_scale()
 
 
 def on_open():
     set_start_end_frames()
+    set_base_file_unit_scale()
```

Comment:
Is it dangerous to just change the unit settings for an existing file? I can imagine things might start behaving differently, like a simulation or alike. The user not getting notified on these changes seems quite intrusive and hidden away that whatever they are opening suddenly gets changed?

The same question for `set_start_end_frames` actually - less dangerous likely, but as an artist reopening my scene later I wouldn't want it to be different when opening it again without any notification whatsoever.

Also, should there be a publish validator confirming the state of this setting as well? If someone would've opened the scene, then changed the setting, then saves and publishes it would just pass. But a reopen of the scene would "force it back" potentially also making it very hard to debug why that particular publish might have acted different than when reopening the scene and publishing again (due to it auto-resetting the value without any hint to the user)

Reply:
That's a fair point! I guess it shouldn't really be necessary to change already existing files anyway, as probably they're already set. I'll change that!

Reply:
I added a setting to make it optional to change the unit scale on opening. I've also added a warning message if opening a file and the unit scale is changed.

Reply:
> Is it dangerous to just change the unit settings for an existing file? I can imagine things might start behaving differently, like a simulation or alike. The user not getting notified on these changes seems quite intrusive and hidden away that whatever they are opening suddenly gets changed?

@BigRoy  here is excerpt directly from Blender doc...should quite safe in that respect
![image](https://github.com/ynput/OpenPype/assets/112623825/5411e22a-c5c9-4b33-bb35-70b333fceaa7)